crates/egui/src/lib.rs

```diff
@@ -125,7 +125,7 @@
 //! loop {
 //!     let raw_input: egui::RawInput = gather_input();
 //!
-//!     let full_output = ctx.run(raw_input, |ctx| {
+//!     let full_output = ctx.run(raw_input, egui::ViewportIdPair::ROOT, |ctx| {
```

Comment:
What do you think about adding the `egui::ViewportIdPair::ROOT` to the `RawInput` instead, perhaps as part of the `struct ViewportInfo` I suggested earlier?

That way a `RawInput` is self-descriptive, saying "This is the input for this viewport"

Reply:
So do you mean like:
```rust
struct ViewportInfo{
    id_pair: ViewportIdPair,
    inner_rect: Option<Rect>,
    outer_rect: Option<Rect>,
}

struct RawInput{
    viewport: ViewportInfo
    ..
}
```

And removing the `ViewportIdPair` from `Context::{begin_frame,  run}`
That will be nice, and no more changes are needed from the user!
But that means the `egui_winit::State::take_egui_input` needs the `ViewportIdPair`